Upgrade EUI to v88.1.0

1Copenut opened the pull request:
## PR change summary

`v87.2.0`⏩`v88.1.0`

⚠️ The biggest thing to QA in this PR is several **breaking changes** to `EuiDescriptionList`.

### Description list `columnWidths` prop

This PR introduces a new `columnWidths` prop and removes several Kibana instances of custom CSS overrides to title/description column widths.

The primary motivation behind this is not just to reduce custom CSS, but also because v88.0.0 introduced an underlying CSS change of `column` description lists to using [`display: grid` CSS](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_grid_layout). The new prop allows us to support existing description list custom widths while not requiring Kibana developers to understand or write grid CSS (except for 1 or two scenarios around `max-width`).

⚠️ **No user-facing UI around column widths should have regressed as a result of these changes. If they have, please let us know in this PR.**

### Description list gutter size changes

The prop `gutterSize` has been renamed to `rowGutterSize` and the default size is now `s` instead of `m`.

The change to `s` from `m` means there is an **expected** smaller gap between list items (see below screenshots):

**Current `EuiDescriptionList` with default `rowGutterSize="s"`**
<img width="753" alt="" src="https://github.com/elastic/kibana/assets/934879/c17aef28-ed3b-40c4-84c3-396e788b13bb">

**Prior `EuiDescriptionList` with default `gutterSize="m"`**
<img width="721" alt="" src="https://github.com/elastic/kibana/assets/934879/84d5f5a2-8fa6-4f99-9dc0-73fd143aa1e1">

If Kibana teams prefer to keep the previous `m` gutter for their instances of `EuiDescriptionList`, you have a couple of options:

1. Let EUI team know in the PR and we can set usage back to what it was before
2. Set `rowGutterSize="m"` yourselves manually

---

## [`88.1.0`](https://github.com/elastic/eui/tree/v88.1.0)

- Added `font.defaultUnits` theme token. EUI component font sizes default to `rem` units - this token allows consumers to configure this to `px` or `em` ([#7133](https://github.com/elastic/eui/pull/7133))
- Updated `EuiDescriptionList` with new `columnWidths` prop ([#7146](https://github.com/elastic/eui/pull/7146))

**Bug fixes**

- Fixed `EuiDataGrid`'s keyboard shortcuts popover display ([#7146](https://github.com/elastic/eui/pull/7146))

**CSS-in-JS conversions**

- Renamed `useEuiFontSize()`'s `measurement` option to `unit` for clarity ([#7133](https://github.com/elastic/eui/pull/7133))

## [`88.0.0`](https://github.com/elastic/eui/tree/v88.0.0)

- Updated `EuiDescriptionList` with a new `columnGutterSize` prop ([#7062](https://github.com/elastic/eui/pull/7062))

**Deprecations**

- Deprecated `EuiSuggest`. We recommend using `EuiSelectable` or `EuiComboBox` instead ([#7122](https://github.com/elastic/eui/pull/7122))
- Deprecated `EuiControlBar`. We recommend using `EuiBottomBar` instead ([#7122](https://github.com/elastic/eui/pull/7122))
- Deprecated `EuiColorStops`. We recommend copying the component to your application if necessary ([#7122](https://github.com/elastic/eui/pull/7122))
- Deprecated `EuiNotificationEvent`. We recommend copying the component to your application if necessary ([#7122](https://github.com/elastic/eui/pull/7122))

**Breaking changes**

- Renamed `EuiDescriptionList`'s `gutterSize` prop to `rowGutterSize` ([#7062](https://github.com/elastic/eui/pull/7062))
- `EuiDescriptionList`'s `rowGutterSize` prop now defaults to a size of `s` (was previously `m`) ([#7062](https://github.com/elastic/eui/pull/7062))

**Accessibility**

- Fixed the dark mode colors of inline `EuiDescriptionListTitle`s to meet WCAG color contrast requirements ([#7062](https://github.com/elastic/eui/pull/7062))

**CSS-in-JS conversions**

- Converted `EuiKeyPadMenuItem` to Emotion; Removed `$euiKeyPadMenuSize` and `$euiKeyPadMenuMarginSize` ([#7118](https://github.com/elastic/eui/pull/7118))

Commit:
Bumping EUI to v88.0.0.

Commit:
Updating rowGutterSize prop in EuiDescriptionList.

cee-chen committed:
Update deprecated `EuiKeyPadMenuItem` className selector

- note: for some reason, `getDOMNode()` is returning an array instead of a single element - appears to be a bug with Enzyme

cee-chen commented on src/plugins/data/public/shard_failure_modal/shard_failure_description.tsx:
```diff
@@ -96,7 +96,7 @@ export function ShardFailureDescription(props: ShardFailure) {
         <EuiDescriptionList
           type="responsiveColumn"
-          gutterSize="s"
+          rowGutterSize="s"
```
We can technically remove these props completely now that `rowGutterSize` defaults to `"s"`.

@kyrspl what was the intention of the breaking `EuiDescriptionList` gutter size change? Do you want all downstream instances in Kibana to adopt the new default gutter size, or should we update old usages to set `<EuiDescriptionList size="m">` to preserve the old gutter size?

kyrspl replied:
Visual preference - if we think it's going to break a lot of instances let's switch it to `m` and we can review later.

1Copenut replied:
@kyrspl and I discussed this as the beginning of my workday Tuesday (Aug 29). We decided it would be better to update the `EuiDescriptionList` props `rowGutterSize` and `columnGutterSize` to have default `m`. I'll update the component in EUI, issue a new pull request, then make a minor point release and fold it into this Kibana upgrade.

That way we're keeping the default the same as was prior for row margins, and honoring the local Kibana overrides by passing in `s`.

/cc @cee-chen

1Copenut replied:
Upstream bump to EUI: https://github.com/elastic/eui/pull/7140

cee-chen committed:
Fix deprecated color stop type

Commit:
Updated Jest snapshots.

Commit:
Merge branch 'eui/v88.0.0' of github.com:1Copenut/kibana into eui/v88.0.0

1Copenut committed:
Merge branch 'main' into eui/v88.0.0

elasticmachine commented:
Pinging @elastic/eui-team (EUI)

davismcphee commented on src/plugins/data/public/shard_failure_modal/shard_failure_description.tsx:
```diff
@@ -96,7 +96,6 @@ export function ShardFailureDescription(props: ShardFailure) {
       <EuiFlexItem grow={false}>
         <EuiDescriptionList
           type="responsiveColumn"
-          gutterSize="s"
```
It looks like the shard failure modal doesn't display correctly with these changes:
<img width="796" alt="shard_modal" src="https://github.com/elastic/kibana/assets/25592674/887e1fa5-72f8-4221-9701-e1231f3bd4be">

Unfortunately it's tricky to test this modal since you need shard failures, but they can be simulated for testing if you want to see it:
- Navigate to Discover and create a data view with the pattern `kibana*`.
- Go to this line in `search_source`: https://github.com/elastic/kibana/blob/501e9bda2a6c17732c525d557b97f12c48dfe345/src/plugins/data/common/search/search_source/search_source.ts#L944
- Paste this before the return line:
```ts
body.query.bool.must.push({
  error_query: {
    indices: [
      {
        name: 'kibana_sample_data_logs',
        shard_ids: [0, 1],
        error_type: 'exception',
        message: 'Testing shard failures!',
      },
    ],
  },
});
```
- Reload the page and you should see a shard failure callout with a "Show details" button to open the modal:
  <img width="1027" alt="no_results" src="https://github.com/elastic/kibana/assets/25592674/5f687565-65be-4431-93c4-1485c1c7dc33">

I'm not sure if this is related to the `gutterSize` change or something else in the upgrade, but we do have some custom styles for this in `src/plugins/data/public/shard_failure_modal/_shard_failure_modal.scss` that I'm guessing are probably related.

1Copenut replied:
@davismcphee The fix for this item has just been pushed to this branch. I tested in Chrome, Firefox, Edge, and Safari at mobile and deskop widths to kick off `dispay: block` and `display: grid` layouts.

elasticmachine commented:
Pinging @elastic/uptime (Team:uptime)

opauloh reviewed (approved):
Cloud Security lgtm

jeramysoucy reviewed (approved):
user_profile.test.tsx changes LGTM!

shahzad31 commented on x-pack/plugins/synthetics/public/apps/synthetics/components/common/monitor_test_result/result_details.tsx:
```diff
@@ -130,7 +130,6 @@ export const StepMetrics = ({ step }: { step: JourneyStep }) => {
   return (
     <EuiDescriptionList
       compressed={true}
-      gutterSize="s"
```
shouldn't this be renamed instead of removing?

1Copenut replied:
Hi @shahzad31. This breaking change had two parts, renaming the prop to `rowGutterSize` and remapping the default gutter size to `s`. That's why I removed the `gutterSize="s"` prop entirely instead of removing it.

alisonelizabeth reviewed (approved):
Deployment Management changes LGTM

1Copenut commented on x-pack/plugins/synthetics/public/apps/synthetics/components/common/monitor_test_result/result_details.tsx:
```diff
@@ -96,7 +96,7 @@ export const TimingDetails = ({ step }: { step: JourneyStep }) => {
   return (
     <EuiDescriptionList
       compressed={true}
-      gutterSize="s"
```
@shahzad31 Yes it should. I realized this morning I missed a changed `rowGutterSize` in `x-pack/plugins/synthetics/.../monitor_test_result/result_details.tsx`. I'll update it when I push the fix for a [visual break Davis called out earlier](https://github.com/elastic/kibana/pull/165047#discussion_r1309434633).

1Copenut commented on x-pack/plugins/monitoring/public/components/no_data/__snapshots__/checker_errors.test.js.snap:
```diff
@@ -26,7 +26,7 @@ Array [
       >
         <dt
-          class="euiDescriptionList__title emotion-euiDescriptionList__title-row-m-normal"
+          class="euiDescriptionList__title emotion-euiDescriptionList__title-row-normal-s"
```
@weltenwort Correct, the snapshots updated because the default size was changed to `s`. This was a conscious decision by EUI to reduce the margin between rows on `EuiDescriptionList`. If there are layouts you want to retain the larger margin, they can be overrode by passing `rowGutterSize="m"` into the component.

Commit:
CSS fix for Discover > Shard Failure modal.

davismcphee reviewed (approved):
Confirmed the shard failure modal is displaying correctly now, and I didn't notice any other UI issues in Disover, LGTM 👍

cee-chen committed:
Merge branch 'main' into eui/v88.0.0

cee-chen committed:
Upgrade EUI to v88.1.0

- to get latest EuiDescriptionList changes

cee-chen committed:
[Discover] Update shard failure description list to use column widths prop

- avoids setting custom CSS entirely

cee-chen committed:
[DataViz] Replace `EuiDescriptionList` custom CSS with prop

+ remove existing Sass file(s) if empty

cee-chen committed:
[ML] Update `EuiDescriptionList` CSS overrides

cee-chen committed:
[Security/Triggers] Update `EuiDescriptionList` custom CSS

- note: switch to `responsiveColumn` is slightly opinionated, and is based on the previous min-width media query set

cee-chen committed:
[Security] Update `EuiDescriptionList` custom CSS

+ remove modifier classes that no longer exist

cee-chen committed:
[Security] Remove `descriptionList-20_80` class

- in favor of `columnWidth` props

- Note: margin-top is already handled by new row gutter size default `s` size, so removing it completely

cee-chen committed:
[Security/Synthetics] Remove now-unnecessary `EuiDescriptionList` margin-top custom CSS

- this is now automatically handled by EuiDescriptionList's row gutter size defaulting to an `s` size

cee-chen committed:
[Synthetics] Replace custom `EuiDescriptionList` CSS with `columnWidths`

cee-chen committed:
[Security/Resolver] Update column `max-widths` usages

- should use `fit-content()` instead, which functions in CSS grid as `max-width` previously did

cee-chen committed:
[Canvas] Remove custom styles for new `columnWidths` prop

cee-chen committed:
[Session View] Remove custom styles for new `columnWidths` prop

+ replace `margin-top` with `row-gap` to account for new grid CSS

cee-chen committed:
[EuiDescriptionList] Update snapshots

cee-chen committed:
[Spaces] Remove enzyme snapshot in favor of RTL snapshot

- goal is to reduce churn whenever eui theme updates, and also prefer RTL over enzyme

cee-chen committed:
Merge branch 'main' into eui/v88.0.0

cee-chen committed:
Merge branch 'main' into eui/v88.0.0

cee-chen committed:
Merge branch 'main' into eui/v88.0.0

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/precommit_hook.js --ref HEAD~1..HEAD --fix'

peteharverson reviewed (approved):
Tested ML changes and LGTM

nikitaindik committed:
Fix EuiDescriptionList usage styles

cee-chen committed:
Merge remote-tracking branch 'upstream/main' into eui/v88.0.0

cee-chen commented:
Hey all - just want to give a heads up, I'm not sure what's going on with the failing serverless tests and I'm pretty sure they're not related to EUI. We'll continue merging main in hopes that they get fixed in main, and talk to KibanaOps if it's not looking to improve, but until then, please ignore any serverless CI failures and focus on `EuiDescriptionList` as the main item to QA in this PR.

nikitaindik reviewed (approved):
Reviewed the Security Solution Rules Management screens. Found a couple display issues. Discussed these with @cee-chen. Fixes for these were merged in with this [commit](https://github.com/elastic/kibana/pull/165047/commits/62cdbad78992a5e08fd2a79c818f206ea6d2c6f0).

Now Security Solution Rules Management part LGTM.

tomsonpl reviewed (approved):
LGTM, Management views look ok to me :) But let's wait for a second pair of eyes too 👍

angorayc reviewed (approved):
Tested locally, LGTM, thank you!

cee-chen committed:
Merge branch 'main' into eui/v88.0.0

weltenwort reviewed (approved):
`monitoring` plugin changes LGTM

cee-chen committed:
Merge branch 'main' into eui/v88.0.0

cee-chen commented:
@elastic/security-threat-hunting-investigations @elastic/kibana-visualizations @elastic/kibana-presentation @elastic/security-defend-workflows - last call for reviews/QA checks, we'll be asking KibanaOps to admin merge by EOD.

As always, however, you're welcome to ping us later for help if you run into issues!

cee-chen committed:
Merge branch 'main' into eui/v88.0.0
